pygatt/backends/bgapi/device.py

```diff
@@ -82,14 +82,22 @@ def char_read_handle(self, handle, timeout=None):
                 self._handle, handle))
 
         self._backend.expect(ResponsePacketType.attclient_read_by_handle)
-        matched_packet_type, response = self._backend.expect_any(
-            [EventPacketType.attclient_attribute_value,
-             EventPacketType.attclient_procedure_completed], timeout=timeout)
-        # TODO why not just expect *only* the attribute value response, then it
-        # would time out and raise an exception if allwe got was the 'procedure
-        # completed' response?
-        if matched_packet_type != EventPacketType.attclient_attribute_value:
-            raise BGAPIError("Unable to read characteristic")
+        success = False
+        while not success:
+            matched_packet_type, response = self._backend.expect_any(
+                [EventPacketType.attclient_attribute_value,
+                 EventPacketType.attclient_procedure_completed],
+                timeout=timeout)
+            # TODO why not just expect *only* the attribute value response,
+            # then it would time out and raise an exception if allwe got was
+            # the 'procedure completed' response?
+            if matched_packet_type != EventPacketType.attclient_attribute_value:
+                raise BGAPIError("Unable to read characteristic")
+            if response['atthandle'] == handle:
```

Comment:
Shouldn't an `else` statement sending the value to the callback function of the subscribed channel be included here?